contracts/beyondFramework/dca/base/DCABaseUpgradeableCutted.sol

```diff
@@ -730,19 +746,31 @@ abstract contract DCABaseUpgradeableCutted is
     }
 
     function setSwapPath(
-        address[] memory depositToBluechip,
-        address[] memory bluechipToDeposit
+        address[] memory depositToBluechipPath,
+        address[] memory bluechipToDepositPath,
+        uint256[] memory depositToBluechipBins,
+        uint256[] memory bluechipToDepositBins
     ) public onlyOwner {
         require(
-            depositToBluechip[0] ==
-                bluechipToDeposit[bluechipToDeposit.length - 1] &&
-                depositToBluechip[depositToBluechip.length - 1] ==
-                bluechipToDeposit[0],
+            depositToBluechipPath[0] ==
+                bluechipToDepositPath[bluechipToDepositPath.length - 1] &&
+                depositToBluechipPath[depositToBluechipPath.length - 1] ==
+                bluechipToDepositPath[0],
             "Invalid swap path"
         );
+        require(
```

Comment:
These sanity checks along with `emergencyWithdrawFunds` are based on a restriction that we'll use TraderJoe V2 only. Will this hold for every tokens that we'll DCA into?

Reply:
yeah, this would require even V1 to provide some bins even though they are ignored. We would somehow make our swaplibs more general and add a method for AMM specific checks in the future.

Reply:
If we'll use TJ V2 only, then we can set bin step 0 to use V1. So it's fine for DCA at least for now. But I need to specify multiple DEXs so I'll find a way to solve this in my PR.

For other chains, maybe it's better to make another SwapLib since they will have different interface.